internals/webhooks/subnamespace_webhook.go

```diff
@@ -286,31 +286,42 @@ func ValidateCreateSnsRequest(sns *utils.ObjectContext, parentQuotaObj *utils.Ob
 	return true, ""
 }
 
-// ValidateAllResourceQuotaParamsValid validates that all quota params: storage, cpu, memory, gpu exists and are positive
+// ValidateAllResourceQuotaParamsValid validates that in subnamespace all quota params: storage, cpu, memory, gpu exists and are positive
+// in reasourcepool validates that upper resource pool cannot be created with an empty quota
 // if one of the params is not valid, the function returns false with the relevant error
 func ValidateAllResourceQuotaParamsValid(sns *utils.ObjectContext) (bool, error) {
 	quotaSNS := utils.GetSnsQuotaSpec(sns.Object).Hard
 	resourceQuotaParams := danav1.ZeroedQuota.Hard
 	var missinsResources []string
 	var negativeResources []string
-	for res := range resourceQuotaParams {
-		requestedRes := quotaSNS[res]
-		if requestedRes.Format == "" {
-			missinsResources = append(missinsResources, res.String())
+	if utils.GetSnsResourcePooled(sns.Object) == "true" {
+		isUpperRp, _ := utils.IsUpperResourcePool(sns)
+		if isUpperRp {
+			if len(quotaSNS) == 0 {
+				return false, errors.New(denyMessageUpperRpMustHaveQuota)
+			}
```

Comment:
Are all these `if` statements needed? Think we can combine a few of them using `&&` and lose some scopes